cfme/infrastructure/pxe.py

```diff
@@ -257,52 +369,115 @@ def get_pxe_image_type(self, image_name):
 class PXEServerAll(CFMENavigateStep):
+    VIEW = PXEServersView
     prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
 
     def step(self):
         self.prerequisite_view.navigation.select('Compute', 'Infrastructure', 'PXE')
-        acc.tree("PXE Servers", "All PXE Servers")
+        self.view.sidebar.servers.tree.click_path('All PXE Servers')
 
 
 @navigator.register(PXEServer, 'Add')
 class PXEServerAdd(CFMENavigateStep):
+    VIEW = PXEServerAddView
     prerequisite = NavigateToSibling('All')
 
     def step(self):
-        cfg_btn('Add a New PXE Server')
+        self.prerequisite_view.toolbar.configuration.item_select('Add a New PXE Server')
 
 
 @navigator.register(PXEServer, 'Details')
 class PXEServerDetails(CFMENavigateStep):
+    VIEW = PXEServerDetailsView
     prerequisite = NavigateToSibling('All')
 
     def step(self):
-        acc.tree("PXE Servers", "All PXE Servers", self.obj.name)
+        self.prerequisite_view.sidebar.servers.tree.click_path('All PXE Servers', self.obj.name)
 
 
 @navigator.register(PXEServer, 'Edit')
 class PXEServerEdit(CFMENavigateStep):
+    VIEW = PXEServerEditView
     prerequisite = NavigateToSibling('Details')
 
     def step(self):
-        cfg_btn('Edit this PXE Server')
+        self.prerequisite_view.toolbar.configuration.item_select('Edit this PXE Server')
+
+
+class PXECustomizationTemplatesView(PXEMainView):
+    """
+    represents Customization Template Groups page
+    """
+    @property
+    def is_displayed(self):
+        return (super(self, PXECustomizationTemplatesView).is_displayed and
+                self.entities.title.text == 'All Customization Templates - System Image Types')
+
+
+class PXECustomizationTemplateDetailsView(PXEMainView):
+    """
+    represents some certain Customization Template Details page
+    """
+    toolbar = View.nested(PXEDetailsToolBar)
+
+    @View.nested
+    class entities(View):  # noqa
+        basic_information = SummaryTable(title="Basic Information")
+        script = ScriptBox(locator='//textarea[contains(@id, "script_data")]')
+
+    @property
+    def is_displayed(self):
+        title = 'Customization Template "{name}"'.format(self.context['object'].name)
+        return (super(self, PXECustomizationTemplateDetailsView).is_displayed and
+                self.entities.title.text == title)
+
+
+class PXECustomizationTemplateAddView(View):
+    title = Text('//div[@id="main-content"]//h1')
+    flash = FlashMessages('.//div[@id="flash_msg_div"]/div[@id="flash_text_div" or '
+                          'contains(@class, "flash_text_div")]')
+    name = Input(id='name')
+    description = Input(id='description')
+    image_type = BootstrapSelect(id='img_typ')
+    type = BootstrapSelect(id='typ')
+    script = ScriptBox(locator='//textarea[contains(@id, "script_data")]')
+
+    add = Button('Add')
+    cancel = Button('Cancel')
+
+    @property
+    def is_displayed(self):
+        return False
```

Comment:
no, there are some things which can be used as keys. However, I don't want to define is_displayed for forms because it will be used in am_I_here. We experienced issues several times due to that and there was a decision not to define is_displayed for forms.
For example, off the top of my mind, one of issues was that you cannot get to the same form by doing navigate_to two times.